.github/workflows/status.yaml

```diff
@@ -0,0 +1,64 @@
+name: Custom GH Status from Workflow Artifacts
+
+on:
+  workflow_run:
+    workflows: ["pr"]
+    types:
+      - completed
+
+jobs:
+  process_artifacts:
+    if: ${{ github.event.workflow_run.conclusion == 'success' }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Download artifact
+        uses: actions/download-artifact@v4
+        with:
+          name: gh-status
+          path: downloaded_artifacts/
+          run-id: ${{ github.event.workflow_run.id }}
+          github-token: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
What happens if the `gh-status` artifact doesn't exist? This will be the case for most events that trigger this workflow.

Will this download artifact step fail? If it doesn't, presumably the subsequent steps will.

This should be investigated and handled correctly.

Reply:
The `download-artifact` step itself will fail if the artifact doesn't exist. Do we want it to not fail?